### PR TITLE
Add julia package

### PR DESCRIPTION
**Pull request by sandinv**
Fixes:

Related:

#### For new package PRs only

- [ ] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [X] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [X] REQUIRED - The version of the package is still receiving security updates
- [X] This PR links to the upstream project's support policy (e.g. `endoflife.date`)

### CODE REVIEW

**Comment by sandinv** [comment]
I watched the[ recorded live](https://www.youtube.com/watch?v=-EXcURRowtU) stream for Julia but I couldn't find the packages in the repository

File: julia.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  copyright:
+    - license: MIT
+environment:
+  contents:
+    packages:
+      - build-base
+      - busybox
+      - ca-certificates-bundle
+      - perl
+      - python3
```
Comment (debasishbsws) [inline julia.yaml:15]:
```suggestion
      - python3
      - llvm16-dev
```

You need `LLVM` to build julia.

And for lining you can use https://github.com/chainguard-dev/yam
It should look like
```YAML
package:
  name: julia
  version: 1.9.4
  epoch: 0
  description: The Julia Programming Language
  copyright:
    - license: MIT

environment:
  contents:
    packages:
      - build-base
      - busybox
      - ca-certificates-bundle
      - llvm16-dev
      - perl
      - python3

pipeline:
  - uses: git-checkout
    with:
      repository: https://github.com/JuliaLang/julia.git
      tag: v${{package.version}}
      depth: 1
      expected-commit: 8e5136fa2979885081cd502d2210633dff1d2a1a

  - uses: autoconf/make

  - runs: |
      make install DESTDIR=${{targets.destdir}} prefix=""

  - uses: strip

update:
  enabled: true
  github:
    identifier: JuliaLang/julia
    strip-prefix: v
```

Comment (sandinv) [reply julia.yaml:15]: thanks a lot for your support. Let me update the PR.

**Comment by rawlingsj** [comment]
Arm build is failing - needs investigation.  Maybe @Elizafox might be interested after Rust?

**Comment by rawlingsj** [comment]
I did some work on this.  After updating to 1.10.0 the build intermittently hangs when "Collecting and executing precompile statements".  I've added some logging to the melange pipeline which you can see in the GitHub Action build logs above.  

Additionally when it does work artifacts are written to `/home/build/julia-1.10/**` so we will need to fix this up, possibly taking inspiration from our openjdk packages and how the resulting package layout is organised.

**Comment by rawlingsj** [comment]
Also found a couple of potentially related threads

https://github.com/JuliaLang/julia/issues/33842
https://github.com/JuliaLang/julia/issues/44338

**Pull request closed without merging**